### DISCUSSION TITLE
Matter (Project CHIP) compatible OpenThread firmware for Thread protocol released for Elelabs and Popp Silabs EFR32 based Zigbee adapters

### DISCUSSION THREAD

**Discussion by Hedda** [zigpy/zigpy#850] [General]
Sounds as ELU013 can now be used to make an "OpenThread Border Router" (OTBR) Leader for Matter (Project CHIP) application?

To be honest, have not really read up on how Matter works with a "Thread Border Router Leader" but new firmware should be compatible?

https://github.com/Elelabs/elelabs-zigbee-ezsp-utility/blob/master/data/EFR32MG13/ELE_MG13_ot_rcp_115200_120_211116.gbl

https://github.com/Elelabs/elelabs-zigbee-ezsp-utility/commit/721019cb542b273159372357a30251330e7c8d51

https://github.com/Elelabs/elelabs-zigbee-ezsp-utility/commit/e9b3fe1dd73b46fef1629a7a54ca0a257c1388ae

OpenThread firmware apparently uses "Spinel" (Openthread Serial Protocol) interface instead of the generic Silicon Labs EZSP (EmberZNet Serial Protocol), and "Spinel" is works using OpenThread's "Pyspinel" CLI tool https://github.com/openthread/pyspinel

Elelabs has also made a vendor package for Pyspinel https://github.com/Elelabs/pyspinel/tree/elelabs/elelabs

The reason why posting under zigpy discussion is understand Matter applications and devices also rely on ZCL (Zigbee Cluster Library), or? 

Sadly their new generic brand name "Matter" makes it really hard to do google searches on it so still have to search for "Project CHIP".

https://www.ti.com/lit/pdf/sway034

https://e2e.ti.com/blogs_/b/process/posts/thread-vs-zigbee-what-s-the-difference

Quick summary of Matter over Thread as I understand it is that it will work somewhat similar to a Zigbee network, with the Thread application and devices even depending on the Zigbee Cluster Library(?), but instead of using a Silabs EFR32 based adapter with its EmberZNet Zigbee stack firmware configured as a "Zigbee Coordinator" to control all Zigbee devices on a Zigbee network, users of Thread based Matter devices will need to flash a compatible adapter with an OpenThread firmware and configure that as an "OpenThread Border Router" and make it "Thread Border Router Leader" to make it the media interface to bridge LAN/WiFi-to-Thread for your home automation application software running on a computer to use the Matter connectivity IP-based protocol to communicate with Thread based devices compatible with Matter, (which will practically all battery-operated Matter compatible devices, like sensors such as a temperature sensor, motion sensor, or window and door sensor).

https://staceyoniot.com/thread-101-what-you-need-to-know-about-this-smart-home-protocol-in-2021/

https://staceyoniot.com/should-you-be-concerned-about-the-future-of-zigbee-and-z-wave-due-to-project-chip/

There is also not really any commercial Thread based Matter devices available that you can buy as of yet (though some manufacturers have said that they might offer optional firmware to migrate newer Zigbee devices to Thread). Thus developers currently need to make their own Thread based Matter devices, which is something that could apparently be done on the cheap using Nordic Semiconductor's nRF52840 Dongle and their nRF5 SDK(?):

https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/ug_matter_creating_accessory.html

https://github.com/project-chip/zap

https://www.nordicsemi.com/Products/Development-hardware/nrf52840-dongle/getstarted

Note! Texas Instruments CC2652 and CC1352 are also compatible with OpenThread:

https://training.ti.com/thread-cc2652-cc1352

References:

https://buildwithmatter.com/

https://www.threadgroup.org/news-events/blog/ID/287/Thread-Matter-And-CHIP--This-Glossary-Will-Quickly-Bring-You-Up-To-Speed

https://github.com/project-chip/connectedhomeip

https://github.com/project-chip/connectedhomeip/tree/master/third_party/openthread

https://github.com/project-chip/connectedhomeip/tree/master/third_party

https://github.com/project-chip/connectedhomeip/tree/master/third_party/efr32_sdk

https://github.com/project-chip/connectedhomeip/blob/master/docs/guides/openthread_border_router_pi.md

https://openthread.io/guides/border-router

https://openthread.io/guides/border-router/build

https://openthread.io/codelabs/openthread-border-router

http://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-3244/matter/openthread_border_router_pi.html

http://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-3244/matter/openthread_rcp_nrf_dongle.html

https://www.silabs.com/documents/public/application-notes/an1256-using-sl-rcp-with-openthread-border-router.pdf

**Comment by MattWestb** [comment]
If i have understanding it right is all EFR32MG1X and newer compatible with Open Thread and can being flashed with OTBR or RCP firmware but the MG1X is not having so much flash and ream for running one more advances app like one RGBCT light with all futures Zigbe 3 is adding.

IKEA is famous that there CWS 1 and 2 is having bad behavior then its on the boarder of the hardware can handling (i think they can using using  2 or 3 hardware PWM and need doing the rest with software PWM that is blocking the CPU).

Simple device shall not being any problem porting to EFR332MG1P but not  to demanding ones then running out of resources.

On the other end one boarder router can being one ESP32 with WiFi or Eth and one EFR32MG as IEEE 802.15.4 link and the EFR32MG is only doing the IEEE 802.15.4 communication and the ESP is doing the TCP / PI part and routing between the networks.
